[MTIA] Ensure correct stream behavior for input_buffer add autograd on MTIA

jvandebon opened the pull request:
Test Plan: CI

Differential Revision: D71414498

albanD reviewed (approved):
cc @soulitzer we should do the larger fix!!

jvandebon commented:
@pytorchbot label "topic: not user facing"

jvandebon committed:
[MTIA] Ensure correct stream behavior for input_buffer add autograd on MTIA (#149433)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/149433

Test Plan: CI

Differential Revision: D71414498
